### PR TITLE
Add instructions for setting up Traefik in Docker

### PR DESCRIPTION
**Pull request by Kevinjil**
Include an example which sets up a Traefik reverse proxy in Docker, including ACME for automated https certificate management. The [deprecated and legacy](https://docs.docker.com/network/links/) `--link` flag of run commands is replaced with Docker networks in all run commands.

This example was initially proposed during NWERC 2023, inspired by https://github.com/DOMjudge/domjudge/issues/2177.

### CODE REVIEW

**Review by vmcj — APPROVED** [review]
I'll wait for one additional approval but I like it.

**Review by nickygerritsen — APPROVED** [review]
Also seems good to meAlso seems good to me

**Pull request merged**
